fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTestDataModel.java

```diff
@@ -64,6 +66,7 @@ public class LuceneIndexTestDataModel {
      * </p>
      */
     final Map<Tuple, Map<Tuple, Tuple>> groupingKeyToPrimaryKeyToPartitionKey;
+    Map<Tuple, Function<Message, Message>> updateableRecords;
```

Comment:
This seems very specific and limited - won't it make sense to store the records as "saved records" and handle the update-specific logic in the test class rather than at the model?

Reply:
I may be hyper-focused on the maintenance tests which have less interesting record shapes, but I feel like, if we want to pull this out, I would also want to abstract away the types a little bit, so that the code above doesn't have to worry about whether it is synthetic or not.
Or perhaps, that the `Function` here should be replaced with an object, that has a variety of operations.

Reply:
For the sake of making progress, then, would you like to check this in as-is and postpone these improvements to a later pr?

Reply:
I changed this to be an object in https://github.com/FoundationDB/fdb-record-layer/pull/3003 but I think there is definitely room to grow, and polish there.